### DISCUSSION TITLE
How to stop the library from repeating its message

### DISCUSSION THREAD

**Discussion by HeikkiHietala** [earlephilhower/ESP8266Audio#584] [General]
Hi all,

I used this library to create a talking clock, and it works fine - it does recite the time. The time is from NTP using the ezTime library, and the sound MP3s are made using a robotic emulator. 

My issue with it is that it keeps repeating the time, and I can't get it to just say the "fourteen twenty" once. It keeps saying it until the minute turns and it gets to say "fourteen twenty-one". 

I have tried to install delay into it so as to have it say the time only once a minute, but any delay in any part of the loop stops the MP3 file from playing. Currently I am looking into having an ultrasound distance sensor turn the volume off if there is no one near, but that is a hack at best.

I'll attach my code here, if someone has the time to look into it.

[ESP32_I2SAUDIO_EZTIME_FINAL.zip](https://github.com/earlephilhower/ESP8266Audio/files/10002743/ESP32_I2SAUDIO_EZTIME_FINAL.zip)

**Comment by HeikkiHietala** [comment]
I wonder if there is anything that can be done on this?

**Comment by ivans805** [comment]
maybe make a flag that checks if the time has changed and then recite the time only once
